packages/batch-submitter/src/utils/tx-submission.ts

```diff
@@ -55,6 +70,7 @@ export const submitTransactionWithYNATM = async (
     maxGasPrice: ynatm.toGwei(config.maxGasPriceInGwei),
     gasPriceScalingFunction: ynatm.LINEAR(config.gasRetryIncrement),
     delay: config.resubmissionTimeout,
+    rejectImmediatelyOnCondition: ynatmRejectOn,
```

Comment:
Is the idea that the batch submitter will catch this exception and try again?

Reply:
Based on the test, that looks like it is the case

Reply:
It will abort the retry loop on either reverts or nonce errors so that we can try submitting the full batch again. This will update the nonce.